### DISCUSSION TITLE
Configuration for External Multi-Factor Provider plug

### DISCUSSION THREAD

**Discussion by ImanMo93** [neos-sdi/adfsmfa#343] [General]
Hello Github!

We have an SMS provider with all of the required API. How can I config the adfsmfa to send OTP via SMS?
I can see the External Multi-Factor Provider (Plug) under the MFA Providers, but I don't have a clue how to get on this one.

I'd appreciate any help.
Thank you all in advance

**Comment by redhook62** [comment]
Hi @ImanMo93 

It's relatively simple, please look at the examples provided (Neos.IdentityServer.Multifactor.Samples).
There is an example, single or bidirectional sending.
You must develop this extension, then you can indicate the assembly and type to use.

regards

**Reply by ImanMo93** [reply]
Thanks for the reply!
I found that on the [Wiki ](https://github.com/neos-sdi/adfsmfa/wiki/08D-MFA%20External%20Provider)page!
But the question is, how can I develop this?
I'm not a developer, I can use PowerShell. But, Do we need someone to make it work with Visual Basic?

**Reply by redhook62** [reply]
Yes, you can develop this extension with Visual Studio (2017/2019/2022) in Visual Basic.Net but preferably with C#
